### ISSUE TITLE
ICE 1107 error with webcam on clean Ubuntu 20.04 due to libcurl3 / kurento / gstream error

### ISSUE THREAD

**Issue by jona-es** [bigbluebutton/bbb-install#544]
Sorry if this is not the right place to put this issue.
I've just installed bbb using `wget -qO- https://ubuntu.bigbluebutton.org/bbb-install-2.5.sh | bash -s -- -w -a -v focal-250 -s sub.domain.tld -e info@domain.tld` on a **blank** **ubuntu 20.04 LTS** system which runs as a virtual machine on ESXi. This virtual machine is configured as an exposed host.
I can access the bbb demo easily and I am able to talk and hear others but I'm not able to neither share my screen nor share my webcam. For both I get the ICE 1107 error message. 
In the server logs, e.g. for `journalctl -f -u bbb-html5-*` I can't see anything special, only INFO messages.

Here my details:
Ubuntu 20.04

```
sudo bbb-conf --status
nginx —————————————————► [✔ - active]
freeswitch ————————————► [✔ - active]
redis-server ——————————► [✔ - active]
bbb-apps-akka —————————► [✔ - active]
bbb-fsesl-akka ————————► [✔ - active]
tomcat9 ———————————————► [✔ - active]
mongod ————————————————► [✔ - active]
bbb-html5 —————————————► [✔ - active]
bbb-webrtc-sfu ————————► [✔ - active]
kurento-media-server ——► [✔ - active]
bbb-html5-backend@1 ———► [✔ - active]
bbb-html5-backend@2 ———► [✔ - active]
bbb-html5-frontend@1 ——► [✔ - active]
bbb-html5-frontend@2 ——► [✔ - active]
etherpad ——————————————► [✔ - active]
bbb-web ———————————————► [✔ - active]
bbb-pads ——————————————► [✔ - active]
```

```
BigBlueButton Server 2.5.4 (3063)
                    Kernel version: 5.15.0-41-generic
                      Distribution: Ubuntu 20.04.4 LTS (64-bit)
                            Memory: 16387 MB
                         CPU cores: 4
...

# Potential problems described below

# Warning: The API demos are installed and accessible from:
# ...
```

##################
EDIT:

I 've currently seen that my **kurento-medai-server** seems to have a problem. This is the error.

```
systemctl status kurento-media-server.service | more
● kurento-media-server.service - Kurento Media Server daemon
     Loaded: loaded (/lib/systemd/system/kurento-media-server.service; enabled; vendor preset: enabled)
     Active: active (running) since Fri 2022-07-22 20:34:16 CEST; 11min ago
    Process: 47643 ExecStartPre=/bin/rm -f /var/kurento/.cache/gstreamer-1.5/registry.x86_64.bin (code=exited, status=0/SUCCESS)
   Main PID: 47653 (kurento-media-s)
      Tasks: 17 (limit: 19115)
     Memory: 8.2M
     CGroup: /system.slice/kurento-media-server.service
             └─47653 /usr/bin/kurento-media-server --gst-debug-level=3 --gst-debug=3,Kurento*:4,kms*:4,KurentoWebSocketTransport:5

Jul 22 20:34:16 server systemd[1]: Starting Kurento Media Server daemon...
Jul 22 20:34:16 server systemd[1]: Started Kurento Media Server daemon.
Jul 22 20:34:16 server gst-plugin-scan[47664]: Failed to load plugin '/usr/lib/x86_64-linux-gnu/gstreamer-1.5/libgstcurl.so': /lib/x86_64-linux-gnu/libcurl.so.4: version `CURL_OPENSSL_3' not found (required by /usr/lib/x86_64-linux-gnu/gstreamer-1.5/libgstcurl.so)
```

So I tried to install libcurl3 as mentioned in some external threads, but:
```
Package libcurl3 is not available, but is referred to by another package.
This may mean that the package is missing, has been obsoleted, or
is only available from another source
However the following packages replace it:
  libcurl4:i386 libcurl4
```

Even the unsupported (for 20.04) package for libcurl3 holds `libcurl.so.4`

So here we've got a problem! The installation script is meant for Ubuntu20.04 but obviously it can't work with the sources given by this OS, or let's say there is a conflict between which is linked and which can be given.

How can I get this Gstream / libcurl / kurento / problem fixed?
Furthermore I think nobody will be able to use the installation script on a clean ubuntu as I wasn't either.

**Comment by jona-es** [comment]
I 've currently seen that my kurento-medai-server seems to have a problem. This is the error.

```
systemctl status kurento-media-server.service | more
● kurento-media-server.service - Kurento Media Server daemon
     Loaded: loaded (/lib/systemd/system/kurento-media-server.service; enabled; vendor preset: enabled)
     Active: active (running) since Fri 2022-07-22 20:34:16 CEST; 11min ago
    Process: 47643 ExecStartPre=/bin/rm -f /var/kurento/.cache/gstreamer-1.5/registry.x86_64.bin (code=exited, status=0/SUCCESS)
   Main PID: 47653 (kurento-media-s)
      Tasks: 17 (limit: 19115)
     Memory: 8.2M
     CGroup: /system.slice/kurento-media-server.service
             └─47653 /usr/bin/kurento-media-server --gst-debug-level=3 --gst-debug=3,Kurento*:4,kms*:4,KurentoWebSocketTransport:5

Jul 22 20:34:16 server systemd[1]: Starting Kurento Media Server daemon...
Jul 22 20:34:16 server systemd[1]: Started Kurento Media Server daemon.
Jul 22 20:34:16 server gst-plugin-scan[47664]: Failed to load plugin '/usr/lib/x86_64-linux-gnu/gstreamer-1.5/libgstcurl.so': /lib/x86_64-linux-gnu/libcurl.so.4: version `CURL_OPENSSL_3' not found (required by /usr/lib/x86_64-linux-gnu/gstreamer-1.5/libgstcurl.so)
```

So I thought of installing libcurl3 as mentioned in some external threads, but:
```
Package libcurl3 is not available, but is referred to by another package.
This may mean that the package is missing, has been obsoleted, or
is only available from another source
However the following packages replace it:
  libcurl4:i386 libcurl4
```

So here we got a problem! The installation script is meant for Ubuntu20.04 but obviously it caan't work with the sources given by this OS, or let's say there is a conflict of which is linked and which can be given.

**Comment by RokkuCode** [comment]
Hey,
maybe you post the issue here: https://github.com/bigbluebutton/bigbluebutton/issues

**Comment by jona-es** [comment]
I'm not sure if this is a general issue. @ffdixon is already involved.
I've tried to install it a few times, but I always face the same issue.

**Comment by bpetit** [comment]
Hi,

+1 

It seems I have the exact same error, on

```
Distributor ID:	Ubuntu
Description:	Ubuntu 20.04.5 LTS
Release:	20.04
Codename:	focal
```

**Comment by ffdixon** [comment]
We just did a new install on Ubuntu 20.04 using Digital Ocean and it worked.

```
# cat /etc/lsb-release 
DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=20.04
DISTRIB_CODENAME=focal
DISTRIB_DESCRIPTION="Ubuntu 20.04.5 LTS"

# uname -a
Linux droplet-9934 5.4.0-122-generic #138-Ubuntu SMP Wed Jun 22 15:00:31 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux
```
> blank ubuntu 20.04 LTS system which runs as a virtual machine on ESXi

We're thinking the issue is running it under ESXi.  @bpetit Are you installing in this environment?

**Comment by bpetit** [comment]
Hi.

I am. But it is a KVM host, not an ESXi one. 

As a first guess, I'd say it would be interesting to look at the base VM images, rather than the VMM or hypervisor.

In my case this is an ubuntu 20.04 "minimal". Maybe "complete" ubuntu images already include what's needed for bbb-install, which explains you get a clean installation ?

**Comment by ffdixon** [comment]
I would try the complete ubuntu images.  We're seeing no problem installing on Digital Ocean.

**Comment by bpetit** [comment]
Hi !

I tried with the complete image, no good.

To better understand, what's the exact version of ubuntu you had on DO ?

In the mean time I tried to identify the root cause. It seems that kurento media server, needs libcurl3, while BBB needs libcurl4 (at least bbb-conf --check says so). The two are incompatible (from apt point of view) and libcurl3 is not available as a package in 20.04.5.

I managed to successfully start kurento media server, after having hackishly installed libcurl3 (and broke bbb as libcurl4 had to be removed). So it's not satisfying at all, but it seems to indicate that libcurl3 vs libcurl4 is root problem.

Did you have libcurl3 still on your ubuntu VM ? I wonder what was the difference.

**Comment by bpetit** [comment]
For the record I gave a try to bbb-install 2.6, same result.

In the mean time, looking at the [kurento media server documentation](https://doc-kurento.readthedocs.io/en/latest/project/relnotes/v6_18_0.html) I tend to understand that kurento media server doesn't support ubuntu 20.04 but only ubuntu 18.04.

Would it be possible that Digital Ocean packages its ubuntu 20.04 VMs so the context is different than other providers/virtualization platforms ? If so, you may encounter more and more people having an issue with kurento server on ubuntu 20.04.05 and above. What do you think ? Do you have users on ubuntu 20.04 who have no issues, on other providers than DO, which would invalidate this theory ?

Thanks a lot and hope to talk to you soon about this

**Comment by ffdixon** [comment]
> In the mean time, looking at the kurento media server documentation I tend to understand that kurento media server doesn't support ubuntu 20.04 but only ubuntu 18.04.

We built a version of Kurento to run on Ubuntu 20.04 for 2.5, which is the current release, so there should be no problems with the underlying OS.

My suggestion would be to spin up a Ubuntu 20.04 Digital Ocean droplet and let BigBlueButton install there (it should work).  You can then compare what libraries are installed to see the combination that worked on DO.  

Try the above and let us know if that gets you further.

**Comment by bpetit** [comment]
Seems to be fixed with latest bbb-install 2.6 now, on ubuntu 20.04.
